### PR TITLE
State: Add application passwords reducer

### PR DESCRIPTION
**Pull request by tyxla**
This PR adds a simple reducer for application passwords. It basically takes care of:

* Receiving application passwords (also applicable when creating a new one)
* Removing an application password upon deletion

Part of #22912.

To test:
* Checkout this branch
* Verify all tests pass:

```
npm run test-client client/state/application-passwords/test/reducer.js
```
* Fire a `dispatch( { type: 'APPLICATION_PASSWORDS_REQUEST' } )` in your console and after the request completes, fire a `state.applicationPasswords` to verify the application passwords are there.
* Fire a `dispatch( { type: 'APPLICATION_PASSWORD_CREATE', applicationName: 'example' } )` in your console, and after receiving has completed, verify the new app password is there (by firing `state.applicationPasswords`).
* Fire a `dispatch( { type: 'APPLICATION_PASSWORD_DELETE', appPasswordId: 12345 } )` where `12345` is the ID of your app password you want to delete. After the request has finished, fire a `state.applicationPasswords` to verify that password is now gone.

### CODE REVIEW

File: client/state/application-passwords/reducer.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	APPLICATION_PASSWORD_DELETE_SUCCESS,
+	APPLICATION_PASSWORDS_RECEIVE,
+} from 'state/action-types';
+import { createReducer } from 'state/utils';
+
+export default createReducer( [], {
+	[ APPLICATION_PASSWORD_DELETE_SUCCESS ]: ( state, { appPasswordId } ) =>
+		reject( state, { ID: appPasswordId } ),
+	[ APPLICATION_PASSWORDS_RECEIVE ]: ( state, { appPasswords } ) => appPasswords,
+} );
```
Comment (sirreal) [inline client/state/application-passwords/reducer.js]: `createReducer` has some overhead and I don't think it adds any value over a function with a `switch` here. I'd like to discourage its use for simple reducers like this one.

Comment (tyxla) [reply client/state/application-passwords/reducer.js]: Good call - addressed in 3c2eb9927b2c8102ca16822a8f03efad67888f2f.

**Comment by tyxla** [comment]
@sirreal applied your suggestion. Wanna take a final 👁 before I 🚢 ?

**Comment by tyxla** [comment]
Thanks a bunch @sirreal!

**Pull request merged**